extra-files/split_logs.py

```diff
@@ -0,0 +1,101 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+from xml.dom import minidom
+import datetime
+import argparse
+import os
+import re
+
+def get_logs(path):
+    logs = {} # filename -> lines[]
+    for file_name in os.listdir(path):
+        if not file_name.startswith("guest-test-"):
+            continue
+        with open(path + file_name, 'r') as f:
+            logs[file_name] = f.readlines()
+    return logs
+
+def filter_logs_by_time(logs, start_time, end_time):
+    filtered_logs = {} # filename -> lines[]
+    timestamp_re = r'[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}'
+
+    def line_in_scope(line):
+        timestamp_match = re.search(timestamp_re, line)
+        if timestamp_match is None:
+            return False
+        else:
+            timestamp = datetime.datetime.fromisoformat(
+                timestamp_match.group(0))
+            return start_time <= timestamp <= end_time
+
+    for log in logs:
+        lines = logs[log]
+        lines_to_keep = list(filter(line_in_scope, lines))
+        if len(lines_to_keep) > 0:
+            filtered_logs[log] = lines_to_keep
+
+    return filtered_logs
+
+def main():
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('--junit-xml', required=True,
+                        help='Junit XML file containing test data.')
+    parser.add_argument('--xen-logs',
+                        default='/var/log/xen/console',
+                        help='Where /var/log/xen/console is located\
+                            (useful for testing in dev vm).')
+    parser.add_argument('--outdir', required=True,
+                        help='Where to output the split logs.')
+
+    args = parser.parse_args()
+    os.makedirs(args.outdir, exist_ok=True)
+
+    with open(args.junit_xml, 'r') as f:
+        # split XML since nose2 malformats XML with multiple different root
+        # elements.
+        testsuites_XMLs = re.findall(r"(<testsuite.*?</testsuite>)",
+                                     f.read(), re.DOTALL)
+
+    for testsuite_xml in testsuites_XMLs:
+        xml = minidom.parseString(testsuite_xml)
+        logs = get_logs(args.xen_logs)
+
+        testcases = xml.getElementsByTagName('testcase')
+        for testcase in testcases:
+            status = testcase.attributes['status'].value
+            if status in ["success", "skipped"]:
+                continue
+            name  = testcase.attributes['classname'].value
+            title = testcase.attributes['name'].value
+            title_short = re.match(r'test_[0-9]*', title).group(0)
+
+            time = datetime.timedelta(
+                seconds=float(testcase.attributes['time'].value))
+            timestamp_str = testcase.attributes['timestamp'].value
+            start_time = datetime.datetime.fromisoformat(timestamp_str)
```

Comment:
Attach UTC timezone to it, to be comparable with timezone aware objects (see above). For example with `.replace(tzinfo=datetime.timezone.utc)`.

Reply:
addressed in https://github.com/QubesOS/openqa-tests-qubesos/pull/17/commits/46f1fcc51d7cfe580c1bf50cfd3f816a1df85280